### PR TITLE
[BEAM-5898] Upgrading gRPC to 1.27

### PR DESCRIPTION
**Pull request by suztomo**
Upgrading the following libraries:
- gRPC version to 1.27.2 (latest). No backward incompatible change since 1.25 as per [its release note](https://github.com/grpc/grpc-java/tree/v1.27.2).
- gax to 1.54.0 (latest) that matches gRPC 1.27 ([gax-grpc:1.54.0](https://search.maven.org/artifact/com.google.api/gax-grpc/1.54.0/jar)). Gax-grpc and grpc work together.
- google-cloud-bigquerystorage to 0.125.0-beta (latest). Its grpc dependency matches grpc 1.27 ([google-cloud-bigquerystorage-parent:0.125.0-beta](https://search.maven.org/artifact/com.google.cloud/google-cloud-bigquerystorage-parent/0.125.0-beta/pom))

# Linkage Checker result 

https://gist.github.com/suztomo/dc17e156f095b382c0bd92712edb5c8d

The auto-service-1.0-rc6 has a missing reference but this auto-service artifact is a provided dependency of a provided dependency. This should not cause a problem in Beam runtime.

```
> Class com.google.auto.service.AutoService is not found;
>   referenced by 1 class file
>     com.google.auto.service.processor.AutoServiceProcessor (auto-service-1.0-rc6.jar)
9033a9037,9038
> auto-service-1.0-rc6.jar is at:
>   org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.21.0-SNAPSHOT (compile) / com.google.cloud.bigtable:bigtable-client-core:1.13.0 (compile) / com.google.cloud:google-cloud-bigtable:1.9.1 (compile) / com.google.auto.value:auto-value:1.7 (provided) / com.google.auto.service:auto-service:1.0-rc6 (provided)
```

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -459,7 +459,7 @@ class BeamModulePlugin implements Plugin<Project> {
         google_auth_library_credentials             : "com.google.auth:google-auth-library-credentials:$google_auth_version",
         google_auth_library_oauth2_http             : "com.google.auth:google-auth-library-oauth2-http:$google_auth_version",
         google_cloud_bigquery                       : "com.google.cloud:google-cloud-bigquery:1.108.0",
-        google_cloud_bigquery_storage               : "com.google.cloud:google-cloud-bigquerystorage:0.120.1-beta",
+        google_cloud_bigquery_storage               : "com.google.cloud:google-cloud-bigquerystorage:0.125.0-beta",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Linkage Checker detected that its grpc-api 1.25 is not compatible with grpc-core 1.27. https://issues.apache.org/jira/browse/BEAM-5898?focusedCommentId=17053644&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17053644

**Comment by suztomo** [comment]
Precommit check with the following 6 commands wanted:
Run Java PostCommit
Run Java HadoopFormatIO Performance Test
Run BigQueryIO Streaming Performance Test Java
Run Dataflow ValidatesRunner
Run Spark ValidatesRunner
Run SQL Postcommit

**Comment by lukecwik** [comment]
Run Java PostCommit

**Comment by lukecwik** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by lukecwik** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by lukecwik** [comment]
Run Spark ValidatesRunner

**Comment by lukecwik** [comment]
Run SQL Postcommit

**Comment by lukecwik** [comment]
R: @lukecwik

**Comment by suztomo** [comment]
[Run Java PreCommit](https://builds.apache.org/job/beam_PreCommit_Java_Commit/10279/console) failed:
```
14:40:11 > Task :sdks:java:io:hadoop-file-system:compileTestJava FAILED
14:40:11 /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit/src/sdks/java/io/hadoop-file-system/src/test/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystemTest.java:69: error: error while writing HadoopFileSystemTest: /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit/src/sdks/java/io/hadoop-file-system/build/classes/java/test/org/apache/beam/sdk/io/hdfs/HadoopFileSystemTest.class: Too many open files
14:40:11 public class HadoopFileSystemTest {
14:40:11        ^
14:40:11 Note: /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit/src/sdks/java/io/hadoop-file-system/src/test/java/org/apache/beam/sdk/io/hdfs/HadoopFileSystemTest.java uses or overrides a deprecated API.
14:40:11 Note: Recompile with -Xlint:deprecation for details.
14:40:11 1 error
14:40:11 
14:40:11 > Task :sdks:java:io:jms:compileTestJava
14:40:11 error: error reading /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit/src/sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java; /home/jenkins/jenkins-slave/workspace/beam_PreCommit_Java_Commit/src/sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java: Too many open files
14:40:11 1 error
```

This seems transient (or irrelevant to this PR).

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
[Java Precommit](https://builds.apache.org/job/beam_PreCommit_Java_Phrase/1827/console) failed at Flink?

```
16:11:57 > Task :runners:flink:1.9:test
16:11:57 
16:11:57 org.apache.beam.runners.flink.translation.wrappers.streaming.io.UnboundedSourceWrapperTest > org.apache.beam.runners.flink.translation.wrappers.streaming.io.UnboundedSourceWrapperTest$ParameterizedUnboundedSourceWrapperTest.testWatermarkEmission[numTasks = 2; numSplits=2] FAILED
16:11:57     org.junit.runners.model.TestTimedOutException at UnboundedSourceWrapperTest.java:354
16:11:59 
```

Checking `./gradlew -p runners/flink check` in my workstation.

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java examples on Dataflow with Java 11

**Comment by lukecwik** [comment]
Run Java_Examples_Dataflow_Java11 PreCommit

**Comment by lukecwik** [comment]
Run Java_Examples_Dataflow_Java11 PostCommit

**Comment by suztomo** [comment]
[Flink test failed](https://builds.apache.org/job/beam_PreCommit_Java_Phrase/1834/testReport/junit/org.apache.beam.runners.flink.translation.wrappers.streaming.io/UnboundedSourceWrapperTest$ParameterizedUnboundedSourceWrapperTest/testWatermarkEmission_numTasks___4__numSplits_4_/) again.

```
org.apache.beam.runners.flink.translation.wrappers.streaming.io.UnboundedSourceWrapperTest$ParameterizedUnboundedSourceWrapperTest.testWatermarkEmission[numTasks = 4; numSplits=4]
```

This test occasionally fails in other PRs: [PR #11045 ](https://builds.apache.org/job/beam_PreCommit_Java_Phrase/1821/) and [PR # 11046](https://builds.apache.org/job/beam_PreCommit_Java_Phrase/1830/).

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
R: @lukecwik 
22 successful checks (and 1 unknown Java 11 test)!

**Pull request merged**